easybuild/easyblocks/c/code_saturne.py

```diff
@@ -0,0 +1,167 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild. If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Code_Saturne,
+implemented as an easyblock.
+
+@author: Metin Cakircali (Juelich Supercomputing Centre, FZJ GmbH)
+"""
+
+import os
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.framework.easyconfig.default import CUSTOM
+from easybuild.tools.filetools import apply_regex_substitutions
+from easybuild.tools.run import run_cmd
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+import easybuild.tools.environment as env
+
+
+class EB_Code_underscore_Saturne(EasyBlock):
+    """Support for building and installing Code_Saturne."""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Extra easyconfig parameters specific to Code_Saturne."""
+        extra_vars = EasyBlock.extra_options(extra_vars)
+
+        # add some custom easyconfig parameters specific to Code_Saturne
+        extra_vars.update({
+            'debug': [False, "Build the debug version.", CUSTOM],
+            'slurm': [False, "Build for the slurm workload manager.", CUSTOM],
+        })
+
+        return extra_vars
+
+    def configure_step(self):
+        """Configure step for Code_Saturne."""
+
+        self.log.info("Configuration step is running...")
+
+        # only use the opt flags
+        env.setvar("CFLAGS", os.environ['OPTFLAGS'])
+        env.setvar("CXXFLAGS", os.environ['OPTFLAGS'])
+        env.setvar("FCFLAGS", os.environ['OPTFLAGS'])
+
+        self.log.info("Running ./sbin/bootstrap ...")
+        cmd = './sbin/bootstrap'
+        run_cmd(cmd, log_all=True, simple=True, log_output=True)
+
+        cmd = ' '.join([
+            self.cfg['preconfigopts'],
+            './configure',
+            '--prefix=' + self.installdir,
+            '--without-modules',
+            self.cfg['configopts'],
+            ])
+
+        if self.cfg['debug']:
+            cmd = ' '.join([cmd, '--enable-debug'])
+
+        (out, _) = run_cmd(cmd, log_all=True, simple=False, log_output=True)
+
+        return out
+
+    def build_step(self):
+        """ Build step for Code_Saturne."""
+
+        paracmd = ''
+        if self.cfg['parallel']:
+            paracmd = "-j %s" % self.cfg['parallel']
+
+        cmd = "%s make %s %s" % (self.cfg['prebuildopts'],
+                                 paracmd, self.cfg['buildopts'])
+
+        (out, _) = run_cmd(cmd, log_all=True, simple=False, log_output=True)
+
+        return out
+
+    def install_step(self):
+        """ Install step for Code_Saturne."""
+
+        cmd = "%s make install %s" % (self.cfg['preinstallopts'],
+                                      self.cfg['installopts'])
+
+        (out, _) = run_cmd(cmd, log_all=True, simple=False, log_output=True)
+
+        return out
```

Comment:
Same here, another reason to just derive from `ConfigureMake` and not define `install_step` here?

Reply:
Sorry, I don't get the idea. Can you please give an example how to derive?
just remove the def?